### PR TITLE
Work on Link and Error components

### PR DESCRIPTION
**Pull request by kayladnls**

### CODE REVIEW

**Comment by kayladnls** [comment]
@rcatlin wanna take a look? This should be good to go.

File: src/Resource/Link/Link.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+     * @param string|null $name
+     * @param string      $href
+     */
+    private function __construct($name, $href)
+    {
+        $this->href = $href;
+        $this->name = $name;
+    }
+
+    public static function createSelf($href, $meta = null)
```
Comment (rcatlin) [inline src/Resource/Link/Link.php]: Missing docblocks.

File: src/Resource/Link/LinkCollection.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+namespace Refinery29\ApiOutput\Resource\Link;
+
+use Refinery29\ApiOutput\Serializer\HasSerializer;
+use Refinery29\ApiOutput\Serializer\Link\LinkCollection as Serializer;
+
+class LinkCollection implements HasSerializer
+{
+    use HasLinks;
+
+    public function getSerializer()
```
Comment (rcatlin) [inline src/Resource/Link/LinkCollection.php]: Doc block.

**Comment by rcatlin** [comment]
:+1:

**Pull request merged**